### PR TITLE
[EXT-48] [Canvas Medical] Create Claim

### PR DESCRIPTION
**Pull request by kubiak85**

### CODE REVIEW

File: extensions/canvasMedical/v1/actions/createClaim/config/fields.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+export const fieldsValidationSchema = z.object({
+  status: z.string(),
+  type: z.record(z.string(), z.any()),
+  patientId: z.string(),
+  created: DateOnlySchema,
+  provider: z.record(z.string(), z.any()),
+  supportingInfo: z.array(z.any()).optional(),
+  diagnosis: z.array(z.any()),
+  insurance: z.array(z.any()),
+  item: z.array(z.any()),
```
Comment (michal-grzelak) [inline extensions/canvasMedical/v1/actions/createClaim/config/fields.ts:91]: I believe that won't work. Look at e.g. `extensions/sendbird/v1/actions/updateCustomerCustomFields/config/fields.ts` to see how to validate JSON fields (there is json validator in `extensions/sendbird/v1/validation.ts`). I think `json` fields are not automatically parsed into objects/arrays, at least that was the case a few weeks ago.

cc @rahulkeerthi to confirm

Comment (kubiak85) [reply extensions/canvasMedical/v1/actions/createClaim/config/fields.ts:91]: Well! Thanks for pointing out this problem. I will try to correct it.

Comment (rahulkeerthi) [reply extensions/canvasMedical/v1/actions/createClaim/config/fields.ts:91]: Yes, that's right @michal-grzelak

File: extensions/canvasMedical/v1/actions/createClaim/config/fields.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import {
+  DateOnlySchema,
+  FieldType,
+  type Field,
+} from '@awell-health/extensions-core'
+import { z, type ZodTypeAny } from 'zod'
+
+export const fields = {
```
Comment (rahulkeerthi) [inline extensions/canvasMedical/v1/actions/createClaim/config/fields.ts:8]: I would recommend adding a comment linking to the source documentation for these fields

File: extensions/canvasMedical/v1/actions/createClaim/config/fields.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    label: 'Type',
+    description:
+      'Accepts a JSON object representing the type compliant with Canvas Medical.',
+    type: FieldType.JSON,
+    required: true,
+  },
+  patientId: {
+    id: 'patientId',
+    label: 'Patient ID',
+    description: `A patient's unique ID.`,
```
Comment (rahulkeerthi) [inline extensions/canvasMedical/v1/actions/createClaim/config/fields.ts:30]: `The patient's unique resource ID (e.g. Patient/865058f6654149bd921264d91519af9e)`

**Review by rahulkeerthi — APPROVED** [review]

**Pull request merged**